tests/unittests/algo/test_axoptimizer.py

```diff
@@ -0,0 +1,263 @@
+"""Perform integration tests for `orion.algo.ax`."""
+import statistics as stats
+
+import pytest
+
+from orion.algo.axoptimizer import has_Ax
+from orion.benchmark.task.base import BenchmarkTask
+from orion.core.utils import backward
+from orion.testing.algo import BaseAlgoTests, phase
+
+if not has_Ax:
+    pytest.skip("skipping Ax tests", allow_module_level=True)
+else:
+    import numpy
+    from botorch.test_functions.multi_objective import BraninCurrin
+    from torch import Tensor
+
+N_INIT = 5
+TOL = 0.1
+
+
+class BraninCurrinTask(BenchmarkTask):
+    r"""Two objective problem composed of the Branin and Currin functions.
+
+    Branin (rescaled):
+
+        f(x) = (
+        15*x_1 - 5.1 * (15 * x_0 - 5) ** 2 / (4 * pi ** 2) + 5 * (15 * x_0 - 5)
+        / pi - 5
+        ) ** 2 + (10 - 10 / (8 * pi)) * cos(15 * x_0 - 5))
+
+    Currin:
+
+        f(x) = (1 - exp(-1 / (2 * x_1))) * (
+        2300 * x_0 ** 3 + 1900 * x_0 ** 2 + 2092 * x_0 + 60
+        ) / 100 * x_0 ** 3 + 500 * x_0 ** 2 + 4 * x_0 + 20
+
+    """
+
+    def __init__(self, max_trials=20):
+        self._branincurrin = BraninCurrin()
+        super().__init__(max_trials=max_trials)
+
+    # pylint: disable=arguments-differ
+    def call(self, x):
+        """Evaluate 2-D branin and currin functions."""
+        _y = self._branincurrin(Tensor(x))
+        return [
+            dict(name="branin", type="objective", value=_y[0].item()),
+            dict(name="currin", type="statistic", value=_y[1].item()),
+        ]
+
+    def get_search_space(self):
+        """Return the search space for the task objective function"""
+        rspace = {"x": "uniform(0, 1, shape=2, precision=10)"}
+
+        return rspace
+
+
+class TestAxOptimizer(BaseAlgoTests):
+    """Test suite for algorithm AxOptimizer"""
+
+    algo_name = "axoptimizer"
+    max_trials = 20
+    config = {
+        "seed": 1234,  # Because this is so random
+        "n_initial_trials": N_INIT,
+        "extra_objectives": set(),
+        "constraints": [],
+    }
+
+    def test_configuration_fail(self):
+        """Test that Ax configuration is valid"""
+        with pytest.raises(AssertionError) as exc_info:
```

Comment:
Is the AssertionError in `axoptimizer.py`? or in `self.create_algo` ?

Reply:
None actually, it comes from Ax's client.create_experiment() when it checks for the string format of outcomes constraints